packages/brick_offline_first/lib/src/offline_queue/offline_queue_graphql_client.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:brick_offline_first/src/offline_queue/graphql_request_sqlite_cache_manager.dart';
+import 'package:brick_offline_first/src/offline_queue/graphql_request_sqlite_cache.dart';
+import 'package:gql_link/gql_link.dart';
+import 'package:gql_exec/gql_exec.dart';
+import 'package:logging/logging.dart';
+
+/// Stores all requests in a SQLite database
+class OfflineQueueGraphqlClient extends Link {
+  /// A DocumentNode GraphQL execution interface
+  /// https://pub.dev/documentation/gql_link/latest/link/Link-class.html
+  final Link _inner;
+
+  final GraphqlRequestSqliteCacheManager requestManager;
+
+  final Logger _logger;
+
+  OfflineQueueGraphqlClient(this._inner, this.requestManager)
+      : _logger = Logger('OfflineQueueHttpClient#${requestManager.databaseName}');
+
+  @override
+  Stream<Response> request(Request request, [NextLink? forward]) async* {
+    final cacheItem = GraphqlRequestSqliteCache(request);
+    _logger.finest('sending: ${cacheItem.toSqlite()}');
+
+    final db = await requestManager.getDb();
+    // Log immediately before we make the request
+    await cacheItem.insertOrUpdate(db, logger: _logger);
+
+    /// When the request is null a generic Graphql error needs to be generated
+    final _genericErrorResponse = Stream.fromIterable([
+      const Response(errors: [GraphQLError(message: 'Unknown error')], data: null)
+    ]);
+
+    try {
+      // Attempt to make Graphql Request, handle it as a traditional response to do check
+      final response = await _inner.request(request).first;
+      // get it back as a stream so that it can be return
+      final streamAsStream = _inner.request(request).asBroadcastStream();
```

Comment:
Does this need to be a broadcast stream? Ideally the stream should only need to be consumed exactly once. 
```suggestion
      final streamAsStream = _inner.request(request);
```